Build issues eframe

The following issue is met when using egui:
﻿2025-07-20T20:22:43.4648101Z ##[group]Run ./build-wasm.sh
./build-wasm.sh
shell: /usr/bin/bash -e {0}
env:
  GITHUB_PAGES: true
🚀 Building WASM Task Manager...
📦 Building WASM package...
Error: `cargo metadata` exited with an error:     Updating crates.io index
error: failed to select a version for `eframe`.
    ... required by package `gh_actions v0.1.0 (/home/runner/work/gh_actions/gh_actions)`
versions that meet the requirements `^0.30` are: 0.30.0

package `gh_actions` depends on `eframe` with feature `wasm_bindgen` but `eframe` does not have that feature.


failed to select a version for `eframe` which could resolve this conflict

Caused by: `cargo metadata` exited with an error:     Updating crates.io index
error: failed to select a version for `eframe`.
    ... required by package `gh_actions v0.1.0 (/home/runner/work/gh_actions/gh_actions)`
versions that meet the requirements `^0.30` are: 0.30.0

package `gh_actions` depends on `eframe` with feature `wasm_bindgen` but `eframe` does not have that feature.


failed to select a version for `eframe` which could resolve this conflict

@claude a New issue has risen :
﻿2025-07-20T20:29:24.7677444Z ##[group]Run ./build-wasm.sh
./build-wasm.sh
shell: /usr/bin/bash -e {0}
env:
  GITHUB_PAGES: true
🚀 Building WASM Task Manager...
📦 Building WASM package...
[INFO]: 🎯  Checking for the Wasm target...
info: downloading component 'rust-std' for 'wasm32-unknown-unknown'
info: installing component 'rust-std' for 'wasm32-unknown-unknown'
[INFO]: 🌀  Compiling to Wasm...
   Compiling autocfg v1.5.0
   Compiling once_cell v1.21.3
   Compiling bytemuck_derive v1.10.0
   Compiling version_check v0.9.5
   Compiling zerocopy v0.8.26
   Compiling lock_api v0.4.13
   Compiling parking_lot_core v0.9.11
   Compiling ahash v0.8.12
   Compiling wasm-bindgen v0.2.100
   Compiling smallvec v1.15.1
   Compiling bytemuck v1.23.1
   Compiling ttf-parser v0.25.1
   Compiling scopeguard v1.2.0
   Compiling js-sys v0.3.77
   Compiling emath v0.30.0
   Compiling ab_glyph_rasterizer v0.1.9
   Compiling ecolor v0.30.0
   Compiling owned_ttf_parser v0.25.0
   Compiling ab_glyph v0.2.31
   Compiling parking_lot v0.12.4
   Compiling profiling v1.0.17
   Compiling log v0.4.27
   Compiling nohash-hasher v0.2.0
   Compiling epaint_default_fonts v0.30.0
   Compiling epaint v0.30.0
   Compiling num-traits v0.2.19
   Compiling litrs v0.4.1
   Compiling document-features v0.2.11
   Compiling egui v0.30.0
   Compiling web-sys v0.3.77
   Compiling web-time v1.1.0
   Compiling wasm-bindgen-futures v0.4.50
   Compiling raw-window-handle v0.6.2
   Compiling percent-encoding v2.3.1
   Compiling static_assertions v1.1.0
   Compiling chrono v0.4.41
   Compiling egui_plot v0.30.0
   Compiling eframe v0.30.0
error: You must enable either the 'glow' or 'wgpu' feature
  --> /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/eframe-0.30.0/src/web/mod.rs:27:1
   |
27 | compile_error!("You must enable either the 'glow' or 'wgpu' feature");
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0433]: failed to resolve: could not find `ActiveWebPainter` in `super`
  --> /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/eframe-0.30.0/src/web/app_runner.rs:44:30
   |
44 |         let painter = super::ActiveWebPainter::new(egui_ctx.clone(), canvas, &web_options).await?;
   |                              ^^^^^^^^^^^^^^^^ could not find `ActiveWebPainter` in `super`
   |
note: found an item that was configured out
  --> /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/eframe-0.30.0/src/web/mod.rs:34:17
   |
34 | pub(crate) type ActiveWebPainter = web_painter_glow::WebPainterGlow;
   |                 ^^^^^^^^^^^^^^^^
note: the item is gated behind the `glow` feature
  --> /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/eframe-0.30.0/src/web/mod.rs:33:7
   |
33 | #[cfg(feature = "glow")]
   |       ^^^^^^^^^^^^^^^^
note: found an item that was configured out
  --> /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/eframe-0.30.0/src/web/mod.rs:39:17
   |
39 | pub(crate) type ActiveWebPainter = web_painter_wgpu::WebPainterWgpu;
   |                 ^^^^^^^^^^^^^^^^
note: the item is gated here
  --> /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/eframe-0.30.0/src/web/mod.rs:38:1
   |
38 | #[cfg(all(feature = "wgpu", not(feature = "glow")))]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0412]: cannot find type `ActiveWebPainter` in module `super`
  --> /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/eframe-0.30.0/src/web/app_runner.rs:12:21
   |
12 |     painter: super::ActiveWebPainter,
   |                     ^^^^^^^^^^^^^^^^ not found in `super`
   |
note: found an item that was configured out
  --> /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/eframe-0.30.0/src/web/mod.rs:34:17
   |
34 | pub(crate) type ActiveWebPainter = web_painter_glow::WebPainterGlow;
   |                 ^^^^^^^^^^^^^^^^
note: the item is gated behind the `glow` feature
  --> /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/eframe-0.30.0/src/web/mod.rs:33:7
   |
33 | #[cfg(feature = "glow")]
   |       ^^^^^^^^^^^^^^^^
note: found an item that was configured out
  --> /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/eframe-0.30.0/src/web/mod.rs:39:17
   |
39 | pub(crate) type ActiveWebPainter = web_painter_wgpu::WebPainterWgpu;
   |                 ^^^^^^^^^^^^^^^^
note: the item is gated here
  --> /home/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/eframe-0.30.0/src/web/mod.rs:38:1
   |
38 | #[cfg(all(feature = "wgpu", not(feature = "glow")))]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Some errors have detailed explanations: E0412, E0433.
For more information about an error, try `rustc --explain E0412`.
error: could not compile `eframe` (lib) due to 3 previous errors
Error: Compiling your crate to WebAssembly failed
Caused by: Compiling your crate to WebAssembly failed
Caused by: failed to execute `cargo build`: exited with exit status: 101
  full command: cd "/home/runner/work/gh_actions/gh_actions" && "cargo" "build" "--lib" "--release" "--target" "wasm32-unknown-unknown"

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -15,7 +15,7 @@ wasm-bindgen = "0.2"
 web-sys = "0.3"
 js-sys = "0.3"
 lazy_static = "1.4"
-eframe = { version = "0.30", default-features = false, features = ["default_fonts"] }
+eframe = { version = "0.30", default-features = false, features = ["default_fonts", "glow"] }
 egui = "0.30"
 egui_plot = "0.30"
 chrono = { version = "0.4", features = ["serde", "wasm-bindgen"] }

